File: leoz-common/src/main/java/org/deku/leoz/rest/entity/zalando/v1/DeliveryOption.java
package org.deku.leoz.rest.entity.zalando.v1;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import org.jetbrains.annotations.NotNull;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;
import java.util.Objects;
import java.util.TimeZone;

@ApiModel(description = "Delivery option for a package. Basic information regarding delivery window, cut off and pic up points.")
@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaResteasyServerCodegen", date = "2017-03-10T11:34:55.297Z")
public class DeliveryOption {

    private String id = null;
    private String cutOff = null;
    private String pickUp = null;
    private String deliveryFrom = null;
    private String deliveryTo = null;

    public DeliveryOption(String id, Date cutOff, Date pickUp, Date deliveryFrom, Date deliveryTo) {
        this.id = id;
        this.cutOff = getISO8601StringForDate(cutOff);
        this.pickUp = getISO8601StringForDate(pickUp);
        this.deliveryFrom = getISO8601StringForDate(deliveryFrom);
        this.deliveryTo = getISO8601StringForDate(deliveryTo);
    }

    public DeliveryOption() {
    }

    /**
     * Delivery option identifier
     **/

    @ApiModelProperty(example = "1", required = true, value = "Delivery option identifier")
    @JsonProperty("id")
    @NotNull
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    /**
     * Cut off time for order modification / cancelation
     **/

    @ApiModelProperty(example = "1730", required = true, value = "Cut off time for order modification / cancelation")
    @JsonProperty("cut_off")
    @NotNull
    public String getCutOff() {
        return cutOff;
    }

    public void setCutOff(String cutOff) {
        this.cutOff = cutOff;
    }

    /**
     * Scheduled pick-up time
     **/

    @ApiModelProperty(example = "1730", required = true, value = "Scheduled pick-up time")
    @JsonProperty("pick_up")
    @NotNull
    public String getPickUp() {
        return pickUp;
    }

    public void setPickUp(String pickUp) {
        this.pickUp = pickUp;
    }

    /**
     * Earliest possible delivery time
     **/

    @ApiModelProperty(example = "1730", required = true, value = "Earliest possible delivery time")
    @JsonProperty("delivery_from")
    @NotNull
    public String getDeliveryFrom() {
        return deliveryFrom;
    }

    public void setDeliveryFrom(String deliveryFrom) {
        this.deliveryFrom = deliveryFrom;
    }

    /**
     * Latest possible delivery time
     **/

    @ApiModelProperty(example = "1730", required = true, value = "Latest possible delivery time")
    @JsonProperty("delivery_to")
    @NotNull
    public String getDeliveryTo() {
        return deliveryTo;
    }

    public void setDeliveryTo(String deliveryTo) {
        this.deliveryTo = deliveryTo;
    }


    @Override
    public boolean equals(java.lang.Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        DeliveryOption deliveryOption = (DeliveryOption) o;
        return Objects.equals(id, deliveryOption.id) &&
                Objects.equals(cutOff, deliveryOption.cutOff) &&
                Objects.equals(pickUp, deliveryOption.pickUp) &&
                Objects.equals(deliveryFrom, deliveryOption.deliveryFrom) &&
                Objects.equals(deliveryTo, deliveryOption.deliveryTo);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, cutOff, pickUp, deliveryFrom, deliveryTo);
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("class DeliveryOption {\n");

        sb.append("    id: ").append(toIndentedString(id)).append("\n");
        sb.append("    cutOff: ").append(toIndentedString(cutOff)).append("\n");
        sb.append("    pickUp: ").append(toIndentedString(pickUp)).append("\n");
        sb.append("    deliveryFrom: ").append(toIndentedString(deliveryFrom)).append("\n");
        sb.append("    deliveryTo: ").append(toIndentedString(deliveryTo)).append("\n");
        sb.append("}");
        return sb.toString();
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
     */
    private String toIndentedString(java.lang.Object o) {
        if (o == null) {
            return "null";
        }
        return o.toString().replace("\n", "\n    ");
    }

    private String getISO8601StringForDate(Date date) {
        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US);
        dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
        return dateFormat.format(date);
    }
}

File: leoz-common/src/main/java/org/deku/leoz/rest/entity/zalando/v1/DeliveryOption.kt
package org.deku.leoz.rest.entity.zalando.v1

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty
import org.deku.leoz.rest.entity.ShortTime

import java.text.SimpleDateFormat
import java.util.*

@ApiModel(description = "Delivery option for a package. Basic information regarding delivery window, cut off and pic up points.")
@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaResteasyServerCodegen", date = "2017-03-10T11:34:55.297Z")
class DeliveryOption {

    /**
     * Delivery option identifier
     */

    @get:ApiModelProperty(example = "228", required = true, value = "Delivery option identifier")
    @get:JsonProperty("id")
    var id: String? = null
    /**
     * Cut off time for order modification / cancelation
     */

    @get:ApiModelProperty(example = "2017-03-16T16:30:00.000Z", required = true, value = "Cut off time for order modification / cancelation")
    @get:JsonProperty("cut_off")
    var cutOff: ShortTime? = null
    /**
     * Scheduled pick-up time
     */

    @get:ApiModelProperty(example = "2017-03-16T17:00:00.000Z", required = true, value = "Scheduled pick-up time")
    @get:JsonProperty("pick_up")
    var pickUp: ShortTime? = null
    /**
     * Earliest possible delivery time
     */

    @get:ApiModelProperty(example = "2017-03-16T17:30:00.000Z", required = true, value = "Earliest possible delivery time")
    @get:JsonProperty("delivery_from")
    var deliveryFrom: ShortTime? = null
    /**
     * Latest possible delivery time
     */

    @get:ApiModelProperty(example = "2017-03-16T21:30:00.000Z", required = true, value = "Latest possible delivery time")
    @get:JsonProperty("delivery_to")
    var deliveryTo: ShortTime? = null

    constructor(id: String, cutOff: ShortTime, pickUp: ShortTime, deliveryFrom: ShortTime, deliveryTo: ShortTime) {
        this.id = id
        this.cutOff = (cutOff)
        this.pickUp = (pickUp)
        this.deliveryFrom = (deliveryFrom)
        this.deliveryTo = (deliveryTo)
    }

    constructor() {}


    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o == null || javaClass != o.javaClass) {
            return false
        }
        val deliveryOption = o as DeliveryOption?
        return id == deliveryOption!!.id &&
                cutOff == deliveryOption.cutOff &&
                pickUp == deliveryOption.pickUp &&
                deliveryFrom == deliveryOption.deliveryFrom &&
                deliveryTo == deliveryOption.deliveryTo
    }

    override fun hashCode(): Int {
        return Objects.hash(id, cutOff, pickUp, deliveryFrom, deliveryTo)
    }

    override fun toString(): String {
        val sb = StringBuilder()
        sb.append("class DeliveryOption {\n")

        sb.append("    id: ").append(toIndentedString(id)).append("\n")
        sb.append("    cutOff: ").append(toIndentedString(cutOff)).append("\n")
        sb.append("    pickUp: ").append(toIndentedString(pickUp)).append("\n")
        sb.append("    deliveryFrom: ").append(toIndentedString(deliveryFrom)).append("\n")
        sb.append("    deliveryTo: ").append(toIndentedString(deliveryTo)).append("\n")
        sb.append("}")
        return sb.toString()
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
     */
    private fun toIndentedString(o: Any?): String {
        if (o == null) {
            return "null"
        }
        return o.toString().replace("\n", "\n    ")
    }

    private fun getISO8601StringForDate(date: Date): String {
        val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US)
        dateFormat.timeZone = TimeZone.getTimeZone("UTC")
        return dateFormat.format(date)
    }
}

